presto-docs/src/main/sphinx/plugin/native-sidecar-plugin.rst

```diff
@@ -0,0 +1,105 @@
+==============
+Native Sidecar
+==============
+
+Using the native sidecar plugin in a native cluster, we extend the capabilities of the cluster by allowing the underlying execution engine to be the source of truth.
+All supported functions, types and session properties are retrieved directly from the sidecar - a specialized worker node equipped with enhanced capabilities.
+Only features supported by the native execution engine are exposed during function resolution, resolving session properties and the types supported.
+
+The native sidecar plugin also provides a native plan checker that validates compatibility between Presto and the native execution engine
+by sending Presto plan fragments to a sidecar endpoint `v1/velox/plan`, where they are translated into Velox compatible plan fragments.
+If any incompatibilities or errors are detected during the translation, they are surfaced immediately, allowing Presto to fail fast before allocating resources or scheduling execution.
+
+To use the sidecar functionalities, at least one sidecar worker must be present in the cluster. The system supports flexible configurations: a mixed setup with both sidecar
+and regular C++ workers, or a cluster composed entirely of sidecar workers.
```

Comment:
Might want to add that, in order to make a worker a sidecar, simply add the properties listed in `Sidecar worker properties`.